crates/evm/fuzz/src/strategies/param.rs

```diff
@@ -224,6 +226,149 @@ pub fn fuzz_param_from_state(
+                    }
+                    // Increase array size.
+                    1 => values.push(new_value(param_type, test_runner)),
+                    // Mutate random array element.
+                    2 => mutate_array(&mut values, param_type, test_runner, state),
+                    _ => unreachable!(),
+                }
+                Some(DynSolValue::Array(values))
+            } else {
+                None
+            }
+        }
+        DynSolValue::FixedArray(mut values) => {
+            if let DynSolType::FixedArray(param_type, _size) = param
+                && !values.is_empty()
+            {
+                mutate_array(&mut values, param_type, test_runner, state);
+                Some(DynSolValue::FixedArray(values))
+            } else {
+                None
+            }
+        }
+        DynSolValue::FixedBytes(word, size) => match test_runner.rng().random_range(0..=4) {
+            0 => Word::flip_random_bit(word, size, test_runner),
+            1 => Word::mutate_interesting_byte(word, size, test_runner),
+            2 => Word::mutate_interesting_word(word, size, test_runner),
+            3 => Word::mutate_interesting_dword(word, size, test_runner),
+            4 => None,
+            _ => unreachable!(),
+        }
+        .map(|word| DynSolValue::FixedBytes(word, size)),
+        DynSolValue::CustomStruct { name, prop_names, tuple: mut values } => {
+            if let DynSolType::CustomStruct { name: _, prop_names: _, tuple: tuple_types }
+            | DynSolType::Tuple(tuple_types) = param
+                && !values.is_empty()
+            {
+                // Mutate random struct element.
+                mutate_tuple(&mut values, tuple_types, test_runner, state);
+                Some(DynSolValue::CustomStruct { name, prop_names, tuple: values })
+            } else {
+                None
+            }
+        }
+        DynSolValue::Tuple(mut values) => {
+            if let DynSolType::Tuple(tuple_types) = param
+                && !values.is_empty()
+            {
+                // Mutate random tuple element.
+                mutate_tuple(&mut values, tuple_types, test_runner, state);
+                Some(DynSolValue::Tuple(values))
+            } else {
+                None
+            }
+        }
+        _ => None,
+    }
+    .unwrap_or_else(|| new_value(param, test_runner))
+}
+
+/// Mutates random value from given tuples.
+fn mutate_tuple(
+    tuples: &mut [DynSolValue],
+    tuple_types: &[DynSolType],
+    test_runner: &mut TestRunner,
+    state: &EvmFuzzState,
+) {
+    let id = test_runner.rng().random_range(0..tuples.len());
+    let param_type = &tuple_types[id];
+    let old_val = replace(&mut tuples[id], DynSolValue::Bool(false));
+    let new_val = mutate_param_value(param_type, old_val, test_runner, state);
+    tuples[id] = new_val;
+}
+
+/// Mutates random value from given array.
+fn mutate_array(
```

Comment:
this mutates a single element, maybe rename to `mutate_array_value`? Because this doesn't cover the mutations inside of `FixedArray`